src/js/media-controller.js

```diff
@@ -472,7 +472,16 @@ class MediaController extends MediaContainer {
           MediaUIAttributes.MEDIA_DURATION,
           getDuration(this)
         );
-        this.propagateMediaState(MediaUIAttributes.MEDIA_STREAM_TYPE);
+      },
+      'emptied,durationchange,loadedmetadata,streamtypechange': () => {
+        this.propagateMediaState(
+          MediaUIAttributes.MEDIA_STREAM_TYPE
+        );
+      },
+      'emptied,durationchange,loadedmetadata,targetlivewindowchange': () => {
+        this.propagateMediaState(
+          MediaUIAttributes.MEDIA_TARGET_LIVE_WINDOW
+        );
```

Comment:
NOTE: On unchanged lines https://github.com/muxinc/media-chrome/pull/476/files#diff-f093c2ba0a65c5a566a4ad86659ffc5bee29bfbbc241f863bed8f634b7a0f9f7R508-R510 where we monitor for events to update `MEDIA_TIME_IS_LIVE`, this does not account for all cases where that value may change, nor is there any set of events on an `HTMLMediaElement` that would catch all cases. Given our current use case, we may want to instead model something like `MEDIA_IS_PLAYING_LIVE` that means:

1. media is currently playing
2. the media playing is within the live edge window

This more circumscribed attribute should be able to be reliably updated based on events and would simplify `media-live-button`.

Reply:
for the updated implementation of `media-live`button` in this PR, the edge case will not be a problem. However, the `media-time-is-live` will "lie" on a subset of initialization cases for live/dvr, particularly when autoplay is off. As long as we're ok with that attribute not always being accurate, we can definitely treat that as out of scope and a "we'll deal with it if/when it becomes an issue" scenario, since it doesn't impact any current media chrome use cases/components.